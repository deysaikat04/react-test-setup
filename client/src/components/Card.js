import React from "react";
const Card = ({ attrs }) => {
  const { ipName, ipId, ipClass, ipSize, ipType, ipPlaceholder } = attrs;
  const str = `<input
  type="${ipType}"
  id="${ipId}"
  className="${ipClass}"
  aria-describedby="ipSize"
  name="${ipName}"
  size="${ipSize}"
  placeholder="${ipPlaceholder}"
/>`;
  return (
    <div className="card text-dark bg-light mt-5">
      <div className="card-body">
        <h5 className="card-title">Code of the dynamic input field</h5>
        <p className="">
          <code>{str}</code>
        </p>
      </div>
    </div>
  );
};

export default Card;
