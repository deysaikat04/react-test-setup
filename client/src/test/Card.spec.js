import React from "react";
import {Provider} from 'react-redux';
import Card from "../components/Card";
import renderer from "react-test-renderer";
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);

const testStoreVal = {
    id:"abcd",
    ipName: "email",
    ipId: "emailId",
    ipClass: "email-class",
    ipSize: "10",
    ipType: "email",
    ipPlaceholder:"Enter your email...",
    ipLabel:"Email",
    error: ""
};

const storeVal = {
    attributes: testStoreVal
};

const mockStoreVal = mockStore(storeVal);

const props = {
    attrs: testStoreVal
}

const rendererCreateComp = () => renderer.create(<Provider store={mockStoreVal}><Card {...props}/></Provider>);

describe("Card", () => {
  it("renders", () => {
    const component = rendererCreateComp();
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


