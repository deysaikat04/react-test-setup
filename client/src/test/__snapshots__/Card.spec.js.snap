// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
<div
  className="card text-dark bg-light mt-5"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Code of the dynamic input field
    </h5>
    <p
      className=""
    >
      <code>
        &lt;input
  type="email"
  id="emailId"
  className="email-class"
  aria-describedby="ipSize"
  name="email"
  size="10"
  placeholder="Enter your email..."
/&gt;
      </code>
    </p>
  </div>
</div>
`;
